cli/src/commands/mod.rs

```diff
@@ -280,27 +281,34 @@ struct RestoreArgs {
     revision: Option<RevisionArg>,
 }
 
-/// Run a command across a set of revisions.
+/// Touch up the content changes in a revision with a diff editor
 ///
+/// With the `-r` option, which is the default, starts a diff editor (`meld` by
+/// default) on the changes in the revision.
 ///
-/// All recorded state will be persisted in the `.jj` directory, so occasionally
-/// a `jj run --clean` is needed to clean up disk space.
+/// With the `--from` and/or `--to` options, starts a diff editor comparing the
+/// "from" revision to the "to" revision.
 ///
-/// # Example
+/// Edit the right side of the diff until it looks the way you want. Once you
+/// close the editor, the revision specified with `-r` or `--to` will be
+/// updated. Descendants will be rebased on top as usual, which may result in
+/// conflicts.
 ///
-/// # Run pre-commit on your local work
-/// $ jj run 'pre-commit.py .github/pre-commit.yaml' -r (main..@) -j 4
-///
-/// This allows pre-commit integration and other funny stuff.
+/// See `jj restore` if you want to move entire files from one revision to
+/// another. See `jj squash -i` or `jj unsquash -i` if you instead want to move
+/// changes into or out of the parent revision.
 #[derive(clap::Args, Clone, Debug)]
-#[command(verbatim_doc_comment)]
-struct RunArgs {
-    /// The command to run across all selected revisions.
-    #[arg(long, short, alias = "x")]
-    command: String,
-    /// The revisions to change.
-    #[arg(long, short, default_value = "@")]
-    revisions: Vec<RevisionArg>,
+struct DiffeditArgs {
+    /// The revision to touch up. Defaults to @ if neither --to nor --from are
+    /// specified.
+    #[arg(long, short)]
+    revision: Option<RevisionArg>,
+    /// Show changes from this revision. Defaults to @ if --to is specified.
+    #[arg(long, conflicts_with = "revision")]
+    from: Option<RevisionArg>,
+    /// Edit changes in this revision. Defaults to @ if --from is specified.
+    #[arg(long, conflicts_with = "revision")]
+    to: Option<RevisionArg>,
 }
```

Comment:
It looks like you added back `DiffeditArgs`, so please delete that again

Reply:
it should be split now, I'll take a look at `DiffeditArgs` tomorrow.